File: docs/test-cases/prompt-injection.py
# Test case: prompt-injection (LLM01:2025)
import anthropic

client = anthropic.Anthropic()

def answer_question(user_question, retrieved_docs):
    # BUG: user input and retrieved docs concatenated directly into system prompt
    system = f"""You are a helpful assistant with access to company documents.

    Context from documents:
    {retrieved_docs}

    The user asked: {user_question}

    Answer based on the documents above."""

    response = client.messages.create(
        model="claude-opus-4-6",
        max_tokens=1024,
        system=system,
        messages=[{"role": "user", "content": "Please help me."}]
    )
    return response.content[0].text

def chat_with_email(email_content, user_instruction):
    # BUG: email content (external, attacker-controlled) mixed with instructions
    prompt = f"User wants you to: {user_instruction}\n\nEmail content:\n{email_content}"
    response = client.messages.create(
        model="claude-opus-4-6",
        max_tokens=512,
        messages=[{"role": "user", "content": prompt}]
    )
    return response.content[0].text

File: docs/test-cases/prompt-injection.go
// Test case: prompt-injection (LLM01:2025)
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"
	"strings"

	"github.com/anthropics/anthropic-sdk-go"
)

var client = anthropic.NewClient()

func translateHandler(w http.ResponseWriter, r *http.Request) {
	userInput := r.URL.Query().Get("text")
	// BUG: user input formatted straight into the instruction tier
	prompt := fmt.Sprintf("Translate the following to French: %s", userInput)
	resp, _ := client.Messages.New(r.Context(), anthropic.MessageNewParams{
		Model:     anthropic.F("claude-opus-4-6"),
		MaxTokens: anthropic.F(int64(512)),
		Messages: anthropic.F([]anthropic.MessageParam{
			anthropic.NewUserMessage(anthropic.NewTextBlock(prompt)),
		}),
	})
	fmt.Fprint(w, resp.Content[0].Text)
}

func ragHandler(w http.ResponseWriter, r *http.Request) {
	var body struct {
		Question string   `json:"question"`
		Docs     []string `json:"docs"`
	}
	json.NewDecoder(r.Body).Decode(&body)

	// BUG: retrieved documents concatenated into the system prompt with no
	// delimiter tags — indirect injection via poisoned vector store content
	system := "You are a support agent. Use this context:\n" + strings.Join(body.Docs, "\n")
	// BUG: user question also splatted into the system role
	system += "\nThe user's question is: " + body.Question

	resp, _ := client.Messages.New(context.Background(), anthropic.MessageNewParams{
		Model:     anthropic.F("claude-opus-4-6"),
		MaxTokens: anthropic.F(int64(1024)),
		System:    anthropic.F([]anthropic.TextBlockParam{anthropic.NewTextBlock(system)}),
		Messages: anthropic.F([]anthropic.MessageParam{
			anthropic.NewUserMessage(anthropic.NewTextBlock("Please answer.")),
		}),
	})
	fmt.Fprint(w, resp.Content[0].Text)
}
